Python/errors.c

```diff
@@ -169,6 +191,16 @@ _PyErr_SetObject(PyThreadState *tstate, PyObject *exception, PyObject *value)
         fixed_value = _PyErr_CreateException(exception, value);
         Py_XDECREF(value);
         if (fixed_value == NULL) {
+            PyObject *exc = _PyErr_GetRaisedException(tstate);
+            assert(PyExceptionInstance_Check(exc));
+
+            PyObject *note = get_normalization_failure_note(tstate, exception, value);
+            if (note != NULL) {
+                PyObject *res = _PyException_AddNote((PyBaseExceptionObject*)exc, note);
+                Py_DECREF(note);
+                Py_XDECREF(res);
+            }
+            _PyErr_SetRaisedException(tstate, exc);
```

Comment:
If get_normalization_failure_note returns NULL then this will overwrite whatever exception came from there (including MemoryError). Should I do something else?